packages/client-react-streaming/src/DataTransportAbstraction/WrappedApolloClient.tsx

```diff
@@ -135,150 +128,68 @@ class ApolloClientClientBaseImpl extends ApolloClientBase {
     TransportIdentifier,
     SimulatedQueryInfo
   >();
-  private transportedQueryOptions = new Map<
-    TransportIdentifier,
-    WatchQueryOptions
-  >();
-
-  protected identifyUniqueQuery(options: {
-    query: DocumentNode;
-    variables?: unknown;
-  }) {
-    const transformedDocument = this.documentTransform.transformDocument(
-      options.query
-    );
-    const queryManager = getQueryManager(this);
-    // Calling `transformDocument` will add __typename but won't remove client
-    // directives, so we need to get the `serverQuery`.
-    const { serverQuery } = queryManager.getDocumentInfo(transformedDocument);
-
-    if (!serverQuery) {
-      throw new Error("could not identify unique query");
-    }
-
-    const canonicalVariables = canonicalStringify(options.variables || {});
-
-    const cacheKeyArr = [print(serverQuery), canonicalVariables];
-    const cacheKey = JSON.stringify(cacheKeyArr);
-
-    return {
-      cacheKey,
-      cacheKeyArr,
-    };
-  }
 
   onQueryStarted({ options, id }: Extract<QueryEvent, { type: "started" }>) {
     const hydratedOptions = deserializeOptions(options);
-    const { cacheKey, cacheKeyArr } = this.identifyUniqueQuery(hydratedOptions);
-    this.transportedQueryOptions.set(id, hydratedOptions);
 
-    const queryManager = getQueryManager(this);
+    const [controller, stream] = getInjectableEventStream();
 
-    if (
-      !queryManager["inFlightLinkObservables"].peekArray(cacheKeyArr)
-        ?.observable
-    ) {
```

Comment:
No more manually added `inFlightLinkObservables` - we really use a link, so this doesn't need to fiddle with `QueryManager` internals.